app/uk/gov/hmrc/eoricommoncomponent/frontend/connector/EnrolmentStoreProxyConnector.scala

```diff
@@ -25,38 +27,35 @@ import uk.gov.hmrc.eoricommoncomponent.frontend.domain.{EnrolmentResponse, Enrol
 import uk.gov.hmrc.eoricommoncomponent.frontend.models.events.EnrolmentStoreProxyEvent
 import uk.gov.hmrc.eoricommoncomponent.frontend.util.HttpStatusCheck
 import uk.gov.hmrc.http.{BadRequestException, HeaderCarrier, HttpClient, HttpResponse}
-
-import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class EnrolmentStoreProxyConnector @Inject() (http: HttpClient, appConfig: AppConfig, audit: Auditable)(implicit
   ec: ExecutionContext
-) {
-
-  private val logger = Logger(this.getClass)
+) extends HandleResponses {
 
   private val baseUrl        = appConfig.enrolmentStoreProxyBaseUrl
   private val serviceContext = appConfig.enrolmentStoreProxyServiceContext
 
-  def getEnrolmentByGroupId(
-    groupId: String
-  )(implicit hc: HeaderCarrier, reads: Reads[EnrolmentStoreProxyResponse]): Future[EnrolmentStoreProxyResponse] = {
+  def getEnrolmentByGroupId(groupId: String)(implicit
+    hc: HeaderCarrier,
+    reads: Reads[EnrolmentStoreProxyResponse]
+  ): EitherT[Future, ResponseError, EnrolmentStoreProxyResponse] = EitherT {
     val url =
       s"$baseUrl/$serviceContext/enrolment-store/groups/$groupId/enrolments?type=principal"
 
     // $COVERAGE-OFF$Loggers
     logger.debug(s"GetEnrolmentByGroupId: $url and hc: $hc")
     // $COVERAGE-ON
 
-    http.GET[HttpResponse](url) map { resp =>
-      logResponse(resp)
+    http.GET[HttpResponse](url) map { response =>
+      logResponse(response)
 
-      val parsedResponse = resp.status match {
-        case OK => resp.json.as[EnrolmentStoreProxyResponse]
+      val parsedResponse = response.status match {
+        case OK => handleResponse[EnrolmentStoreProxyResponse](response)
         case NO_CONTENT =>
-          EnrolmentStoreProxyResponse(enrolments = List.empty[EnrolmentResponse])
+          Right(EnrolmentStoreProxyResponse(enrolments = List.empty[EnrolmentResponse]))
         case _ =>
-          throw new BadRequestException(s"Enrolment Store Proxy Status : ${resp.status}")
+          throw new BadRequestException(s"Enrolment Store Proxy Status : ${response.status}")
```

Comment:
Since we're now using eithers could we pass this back as a left?

Reply:
Same with the recover below?

Reply:
True, added those - should the bottom one be InternalServerError rather than Bad request though?